### ISSUE TITLE
PodioOutput writes additional entries to the output file

### ISSUE THREAD

**Issue by zoujh** [key4hep/k4FWCore#36]
When the `EvtMax` option is set greater than the total event number in the input file(s), the PodioOutput will write 2 more entries than the right number to the output file.

We can simply repeat this problem by the default tests. For example, if we change the EvtMax in [test/k4TestFWCore/options/readExampleEventData.py](https://github.com/key4hep/k4FWCore/blob/master/test/k4TestFWCore/options/readExampleEventData.py#L18) to
```
EvtMax=110,
```
The test will be failed, and 102 entries will be saved in the output file "output_k4test_exampledata_2.root".

I traced a problem in [PodioDataSvc](https://github.com/key4hep/k4FWCore/blob/master/k4FWCore/src/PodioDataSvc.cpp#L71) line 71, where
```
if (m_eventNum++ > m_eventMax) {
```
should be
```
if (++m_eventNum >= m_eventMax) {
```
Then, we can get the right entries. But the test will still be failed when EvtMax is greater than (and even equal to) 100.

**Comment by vvolkl** [comment]
Good catch! Indeed I think we should add more checks for that, right now it is just assumed that the number of events will be correct.

**Comment by jmcarcell** [comment]
I just tested this and it doesn't seem to happen anymore. Closing
